courseware/api.py

```diff
@@ -0,0 +1,46 @@
+"""Courseware API functions"""
+from datetime import datetime, timedelta
+from urllib.parse import urljoin
+import pytz
+import requests
+
+from django.conf import settings
+from oauth2_provider.models import Application, AccessToken, RefreshToken
+from oauthlib.common import generate_token
+
+
+OPENEDX_REGISTER_USER_PATH = "/user_api/v1/account/registration/"
+OPENEDX_REQUEST_DEFAULTS = dict(country="US", honor_code=True)
+
+
+def create_lms_user(user):
+    """Makes a request to create an equivalent user in Open edX"""
+    application = Application.objects.get(name=settings.MITXPRO_OAUTH_APP_NAME)
+    expiry_date = datetime.now(tz=pytz.utc) + timedelta(
+        hours=settings.OPENEDX_TOKEN_EXPIRES_HOURS
+    )
+    access_token, _ = AccessToken.objects.get_or_create(
```

Comment:
Changed. I think my silent assumption was that the refresh token was there in case the access token expired, but since this is sort of a one time use token (I believe we only want to use it for creating the user in edx) I just got rid of the refresh token